src/pipelines/master_pipeline.py

```diff
@@ -16,14 +19,17 @@ def main(argv):
     controls the over-arching implmentation of the algorithms
     '''
     directory, features, algorithms, parameters = argv
+    
+    # Create a numpy random state
+    random_state = np.random.RandomState(parameters['seed'])
 
     #parsing
     print("parsing json data...",file=sys.stderr)
-    clusters, order, data, test_clusters, test_order, test_data, corpusdict = parse_json.main([directory, parameters])
+    clusters, order, data, test_clusters, test_order, test_data, corpusdict = parse_json.main(directory, parameters)
 
     #preprocessing
     print("preprocessing...",file=sys.stderr)
-    vocab, encoder_decoder, lda_model, tf_model, w2v_model = preprocess.main([features, parameters, corpusdict, data])
+    vocab, encoder_decoder, lda_model, tf_model, w2v_model = preprocess.main(features, parameters, corpusdict, data)
 
     #featurization
     print("generating training and testing data...",file=sys.stderr)
```

Comment:
Similar to preprocess, data_gen includes a parameter default 'random_state=np.random' but random_state from the master pipeline is not passed in here.

Since the seed value resides inside the parameters dict, one solution could be to remove random_state from the data_gen_main parameters and set random_state via parameters['seed'] inside datagen_main